pybossa/model/event_listeners.py

```diff
@@ -276,6 +283,8 @@ def on_taskrun_submit(mapper, conn, target):
     is_completed = is_task_completed(conn, target.task_id, target.project_id)
     if is_completed:
         update_task_state(conn, target.task_id)
+        check_and_send_project_progress(target.project_id, conn)
```

Comment:
is this going to get called twice? as task state is updated above would trigger [this](https://github.com/bloomberg/pybossa/pull/457/files#diff-c0665c76342c45d6531b2c03e97ad25cR145)? same at line 341 below.

Reply:
`update_task_state`  talks to database directly, while [this](https://github.com/bloomberg/pybossa/pull/457/files#diff-c0665c76342c45d6531b2c03e97ad25cR145) listens to `Task` Object update and delete, see [Task](https://github.com/bloomberg/pybossa/blob/1e9ec0411a71c1529ba8882c9e1ea8bf5b70d8a4/pybossa/model/task.py#L30). 

`update_task_state` happens when user submits answer, `Task` object changes when project owner changes task redundancy through 'Task browse' page. `update_task_state` does not change `Task` object.